### PR TITLE
Add legacy info, moderinzation topic

### LINKED ISSUE #35504: Updates to UA articles: Legacy version, modernization
Add article on installing the legacy UA tool. Add article about modernizing an app after its upgraded

---
[Associated WorkItem - 95351](https://dev.azure.com/msft-skilling/Content/_workitems/edit/95351)

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Add article on installing the legacy CLI tool
- Add article on modernizing apps to .NET from .NET Framework

Fixes #35504

Depends on dotnet/docs-desktop#1641

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/porting/modernize.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/modernize.md) | [docs/core/porting/modernize](https://review.learn.microsoft.com/en-us/dotnet/core/porting/modernize?branch=pr-en-us-35505) |
| [docs/core/porting/upgrade-assistant-install-legacy.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/upgrade-assistant-install-legacy.md) | [docs/core/porting/upgrade-assistant-install-legacy](https://review.learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-install-legacy?branch=pr-en-us-35505) |
| [docs/core/porting/upgrade-assistant-overview.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/upgrade-assistant-overview.md) | [Overview of the .NET Upgrade Assistant](https://review.learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview?branch=pr-en-us-35505) |
| [docs/core/porting/upgrade-assistant-uwp-framework.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/upgrade-assistant-uwp-framework.md) | [Upgrade UWP apps to .NET 6](https://review.learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-uwp-framework?branch=pr-en-us-35505) |
| [docs/core/porting/upgrade-assistant-wcf.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/upgrade-assistant-wcf.md) | [Upgrade a WCF Server-side Project to use CoreWCF on .NET 6](https://review.learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-wcf?branch=pr-en-us-35505) |
| [docs/core/porting/windows-compat-pack.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/core/porting/windows-compat-pack.md) | [docs/core/porting/windows-compat-pack](https://review.learn.microsoft.com/en-us/dotnet/core/porting/windows-compat-pack?branch=pr-en-us-35505) |
| [docs/whats-new/dotnet-docs-mod0.md](https://github.com/dotnet/docs/blob/774802380edf82f8c3827d174eb6bd3caebca863/docs/whats-new/dotnet-docs-mod0.md) | [.NET docs: What's new for March 2023](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-docs-mod0?branch=pr-en-us-35505) |

### CODE REVIEW

File: docs/core/porting/modernize.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+no-loc: ["package.config", PackageReference]
+---
+
+# Modernize after upgrading to .NET from .NET Framework
+
+In this article, you'll learn about different ways you can modernize your .NET Framework app after it's been upgraded to .NET. Use the [.NET Upgrade Assistant](upgrade-assistant-overview.md) tool to upgrade your app to .NET.
+
+## Missing APIs
+
+When upgrading a .NET Framework app, you'll most likely have some incompatibilities. This is because .NET Framework is a Windows-only technology and .NET is a cross-platform technology. Some libraries aren't For example, .NET doesn't provide out-of-the-box APIs to access the Windows Registry like .NET Framework did. Support for the Windows Registry is provided by the `Microsoft.Win32.Registry` NuGet package. Many the .NET Framework-specific libraries have been ported to .NET or .NET Standard, and are hosted on NuGet. If you find a missing reference in your project, search NuGet.
```
Comment (BillWagner) [inline docs/core/porting/modernize.md:16]: nit (missing period):

```suggestion
When upgrading a .NET Framework app, you'll most likely have some incompatibilities. This is because .NET Framework is a Windows-only technology and .NET is a cross-platform technology. Some libraries aren't. For example, .NET doesn't provide out-of-the-box APIs to access the Windows Registry like .NET Framework did. Support for the Windows Registry is provided by the `Microsoft.Win32.Registry` NuGet package. Many the .NET Framework-specific libraries have been ported to .NET or .NET Standard, and are hosted on NuGet. If you find a missing reference in your project, search NuGet.
```

File: docs/core/porting/upgrade-assistant-install-legacy.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+Choose a command:
+   1. Apply next step (Back up project)
+   2. Skip next step (Back up project)
+   3. See more step details
+   4. Select different project
+   5. Configure logging
+   6. Exit
+```
+
+Each step details first what it's going to do, then prompts you to do it. Any step that doesn't apply, is skipped when the tool arrives to the step. For example, if the tool processes **7. Update WCF service to CoreWCF (Preview)**, but your app doesn't define a WCF service, it's skipped and step 8 is processed. In turn, if step 8 doesn't apply, it's skipped too. You may see many steps skipped as the tool tries to find the next step that applies.
```
Comment (BillWagner) [inline docs/core/porting/upgrade-assistant-install-legacy.md:158]: nit:

```suggestion
Each step first details what it's going to do, then prompts you to do it. Any step that doesn't apply, is skipped when the tool arrives to the step. For example, if the tool processes **7. Update WCF service to CoreWCF (Preview)**, but your app doesn't define a WCF service, it's skipped and step 8 is processed. In turn, if step 8 doesn't apply, it's skipped too. You may see many steps skipped as the tool tries to find the next step that applies.
```

**Review by BillWagner — APPROVED** [review]
This all LGTM @adegeo 

I found a couple nits here, but otherwise, it's ready to merge.

File: docs/core/porting/upgrade-assistant-install-legacy.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+---
+title: How to use the legacy .NET Upgrade Assistant
+description: This article describes how to install the old .NET Upgrade Assistant tool.
+author: adegeo
+topic: how-to
+ms.date: 05/22/2023
+---
+# Install the .NET Upgrade Assistant (legacy)
```
Comment (OliaG) [inline docs/core/porting/upgrade-assistant-install-legacy.md:8]: Should we call this article something like "Legacy version of Upgrade Assistant" not the "Install the .NET Upgrade Assistant (legacy)" since it covers not just installation but also the usage?

File: docs/core/porting/upgrade-assistant-uwp-framework.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ ms.date: 09/20/2022
 # Migrate UWP apps to Windows App SDK with the .NET Upgrade Assistant
```
Comment (OliaG) [inline docs/core/porting/upgrade-assistant-uwp-framework.md:8]: Actually, we do have support for UWP in the new VSIX and new CLI so we need to update the docs to suggest that. We can do it as a separate PR.

**Review by OliaG — COMMENTED** [review]
Looks great, added a few comments

**Review by OliaG — COMMENTED** [review]
Looks great, added a few comments
